nilearn/regions/rena_clustering.py

```diff
@@ -0,0 +1,580 @@
+
+        if self.memory is None or isinstance(self.memory, six.string_types):
+            self.memory_ = Memory(cachedir=self.memory,
+                                  verbose=max(0, self.verbose - 1))
+        else:
+            self.memory_ = self.memory
+
+        if self.n_clusters <= 0:
+            raise ValueError("n_clusters should be an integer greater than 0."
+                             " %s was provided." % str(self.n_clusters))
+
+        if self.n_iter <= 0:
+            raise ValueError("n_iter should be an integer greater than 0."
+                             " %s was provided." % str(self.n_iter))
+
+        if self.n_clusters > n_features:
+            self.n_clusters = n_features
+            warnings.warn("n_clusters should be at most the number of "
+                          "features. Taking n_clusters = %s instead."
+                          % str(n_features))
+
+        n_components, labels = self.memory_.cache(
+            recursive_neighbor_agglomeration)(X, self.mask_img,
+                                              self.n_clusters,
+                                              n_iter=self.n_iter,
+                                              threshold=self.threshold,
+                                              verbose=self.verbose)
+
+        sizes = np.bincount(labels)
+        sizes = sizes[sizes > 0]
+
+        self.labels_ = labels
+        self.n_clusters_ = np.unique(self.labels_).shape[0]
+        self.sizes_ = sizes
+
+        return self
+
+    def transform(self, X, y=None):
+        """Apply clustering, reduce the dimensionality of the data.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Data to transform with the fitted clustering.
+
+        Returns
+        -------
+        X_red : numpy array
+            2D data matrix of shape [n_samples, n_clusters]
+        """
+
+        check_is_fitted(self, "labels_")
+
+        unique_labels = np.unique(self.labels_)
+
+        mean_cluster = []
+        for label in unique_labels:
+            mean_cluster.append(np.mean(X[:, self.labels_ == label], axis=1))
+
+        X_red = np.array(mean_cluster).T
+
+        if self.scaling:
+            X_red = X_red * np.sqrt(self.sizes_)
+
+        return X_red
+
+    def inverse_transform(self, X_red):
+        """Transform the reduced 2D data matrix back to an image in brain
+        space.
+
+        Parameters
+        ----------
+        X_red : numpy array
+            2D data matrix of shape [n_samples, n_clusters]
+
+        Returns
+        -------
+        X_inv : nibabel.Nifti1Image
+            shape: (n_x, n_y, n_z, n_samples)
+        """
+
+        check_is_fitted(self, "labels_")
+
+        _, inverse = np.unique(self.labels_, return_inverse=True)
+
+        if self.scaling:
+            X_red = X_red / np.sqrt(self.sizes_)
+        X_inv = X_red[..., inverse]
+
+        return self.masker_.inverse_transform(X_inv)
```

Comment:
```suggestion
        return X_inv
```

Reply:
X_inv should be a 2D numpy array of shape (n_samples, n_voxels) as all the api works with tables, so that you can return X_inv directly, without using the mask